common/errors.go

```diff
@@ -1958,6 +1968,23 @@ func HasErrorCode(err error, codes ...ErrorCode) bool {
 }
 
 func IsRetryableTowardsUpstream(err error) bool {
+	// Special check for client-side errors that might be “retryable”
+	if e, ok := err.(interface {
+		Code() string
+		Details() map[string]interface{}
+	}); ok {
+		if e.Code() == ErrCodeEndpointClientSideException {
+			// If “retriable: true” is present in the details, treat it as "retryable"
+			if details := e.Details(); details != nil {
+				if r, has := details["retriable"].(bool); has && r {
+					return true
+				}
+			}
+			// Otherwise, it’s not retriable
+			return false
+		}
+	}
```

Comment:
Probably can be replaced by something like:
```go
if rt, ok := err.DeepSearch("retriable").(bool); ok && !rt {
  return false
}
```

Although the `DeepSearch` needs to be updated to return `interface{}` or `any` meaning any type and not only strings!

Reply:
I also suggest to put this lookup 'after' the HasErrorCode logic below, because that one is faster and more often hit vs this one (I'm guessing)